### PR TITLE
Cf yaml

### PR DESCRIPTION
**Pull request by quantumew**
Here is the pull request for the start of the cloudformation stuff. I wanted to do this pull req to make sure I am on the right track considering I am a complete newb at this. There are 2 scripts one builds an archive from the pyshelf repo, copies the decrypted certs and the config to the repo directory, zips it up, and stores it on s3 using server-side encryption (is this sufficient?). The other performs an action on a cloudformation stack (update|create|delete|dry-run) currently dry-run is the only thing I am using to validate the cloudformation config. Once again this is not complete just my progress thus far.

### CODE REVIEW

File: cloudformation/later/shelfIamRole.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html
+Type: AWS::IAM::Role
```
Comment (kyle-long) [inline cloudformation/later/shelfIamRole.yaml]: What is this role for?

File: cloudformation/later/shelfInstanceProfile.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-instanceprofile.html
+Type: AWS::IAM::InstanceProfile
```
Comment (kyle-long) [inline cloudformation/later/shelfInstanceProfile.yaml]: same here

File: cloudformation/shelfSecurityGroupInstance.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            IpProtocol: tcp
+            FromPort: 8080
+            SourceSecurityGroupId:
+                "Fn::GetAtt":
+                    - shelfSecurityGroupLoadBalancer
+                    - GroupId
+            ToPort: 8080
+        -
+            IpProtocol: tcp
+            FromPort: 8443
```
Comment (kyle-long) [inline cloudformation/shelfSecurityGroupInstance.yaml]: I think we may not need this.  I think we should have an ELB that does SSL termination for us.

File: cloudformation/shelfSecurityGroupLoadBalancer.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+Type: AWS::EC2::SecurityGroup
+Properties:
+    GroupDescription: Traffic into and out of the ELB
+    # http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-rule.html
+    SecurityGroupIngress:
+        -
+            CidrIp: 0.0.0.0/0
+            IpProtocol: tcp
+            FromPort: 80
+            ToPort: 80
```
Comment (kyle-long) [inline cloudformation/shelfSecurityGroupLoadBalancer.yaml]: Shouldn't this be 8080?

File: gpg-encrypt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+recipients=(andygertjejansen@outlook.com uilwen@gmail.com)
```
Comment (kyle-long) [inline gpg-encrypt]: This is probably why we should us blackbox specifically.  It will automatically encrypt for everyone.  All admins should have access

Comment (quantumew) [inline gpg-encrypt]: Fixed that already, we are using blackbox_cat. That is vestigal.

**Comment by fidian** [comment]
Should this pull request get merged or closed?

**Comment by quantumew** [comment]
@fidian I am going to delete this repo, it is obsolete now.

**Pull request closed without merging**